engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTextureFromFBO.java

```diff
@@ -193,27 +177,24 @@ public boolean equals(Object other) {
      * binding the appropriate texture to it and configuring a material to use it as input.
      */
     protected class Task implements RenderPipelineTask {
-
-        private int textureSlot;
-        private int textureUnit;
+        private final int textureSlot;
+        private final Material material;
+        private final String materialParameter;
         private int textureId;
-        private Material material;
-        private String parameterName;
 
         /**
          * Constructs an instance of this inner class, according to the given parameters.
          *
          * @param textureSlot an integer indirectly identifying a texture unit on the GPU (textureUnit = GL_TEXTURE0 + textureSlot).
          * @param textureId the opengl id of a texture, usually obtained via glGenTextures().
          * @param materialURN an URN identifying a material.
-         * @param parameterName the name of a variable in the shader program used to sample the texture.
+         * @param materialParameter a string
```

Comment:
I think it's inaccurate. The parameter is really on the shader, not on the material. And it's not really the parameter itself, but its name. I guess the most accurate name would be shaderParameterName. But I thought parameterName + the description was enough.